k8s-api.tf

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ */
+
+locals {
+  api_name      = "api"
+  api_full_name = "${local.k8s_app_alias}-api"
+  api_labels = {
+    "app.kubernetes.io/name"       = local.api_full_name
+    "app.kubernetes.io/instance"   = local.api_full_name
+    "app.kubernetes.io/version"    = var.api_docker_image_version
+    "app.kubernetes.io/managed-by" = local.k8s_app_alias
+  }
+  api_label_name_instance = {
+    "app.kubernetes.io/name"     = local.api_full_name
+    "app.kubernetes.io/instance" = local.api_full_name
+  }
+}
+
+resource "kubernetes_deployment" "beekeeper_api" {
+  count = var.instance_type == "k8s" ? 1 : 0
+  metadata {
+    name      = local.api_full_name
+    namespace = var.k8s_namespace
+    labels    = local.api_labels
+  }
+
+  spec {
+    # setting the number of replicas to greater than 1 is currently untested
+    replicas = 1
+    selector {
+      match_labels = local.api_label_name_instance
+    }
+
+    template {
+      metadata {
+        labels = local.api_label_name_instance
+        annotations = {
+          "iam.amazonaws.com/role" = aws_iam_role.beekeeper_k8s_role_api_iam[count.index].arn
```

Comment:
I think you don't need this IAM role. Can you try remove this line and test

Reply:
If it works, then you can get rid of `k8s-api-iam.tf `

Reply:
true, works without it, took it out